ci/ansible/roles/jenkins-user-add/tasks/main.yml

```diff
@@ -0,0 +1,60 @@
+---
+- name: Create the {{ ci_user }} group
+  group:
+      name: "{{ ci_user }}"
+      state: present
+  become: yes
+
+- name: Create the {{ ci_user }} User
+  user:
+      name: "{{ ci_user }}"
+      comment: Jenkins CI user
+      group: "{{ ci_user }}"
+      password: "{{ ci_password }}"
+  when: ci_password is defined
+  become: yes
+
+- name: Create symbolic link for jenkins agent to user home
+  file:
+    src: /home/{{ ci_user }}/
+    dest: /var/lib/jenkins
+    state: link
+    force: yes
+
+- name: Create {{ ci_user }} .ssh directory 
+  file:
+    path: /home/{{ ci_user }}/.ssh/
+    state: directory
+    mode: "u=rwx"
+    group: "{{ ci_user }}"
+    owner: "{{ ci_user }}"
+    recurse: yes
+  become: yes
+
+- name: Copy {{ ci_user }} User Authorized Keys
+  copy:
+    remote_src: yes
+    src: /home/{{ ansible_user }}/.ssh/authorized_keys
```

Comment:
@kersommoura @rochacbruno please take direct note of which keys this is using.

I am using the ansible_user the (`-u $user`) the plan was run as to reap his authorized_keys (as this is how you got into the machine to run the playbook) and copy those keys to the new `jenkins` user.

If you create the image as our ci-user, this works wonderfully. 

If you create the image as your authenticated user, this will cause the image to NOT work with `pulp-smash` as you copy the wrong keys here.

We can easily patch in a new idea, or just follow the documentation I wrote to get this to work... or just make your private key to SSH into Openstack images jenkin's :)

Reply:
Does it mean that if I run locally it will not work? or I should create a local ansible_user user and place this files on its /home?

Reply:
@rochacbruno depends on the `-u user` you are using to log in. If that user doesn't have the `jenkins.pub`, then yes, it would fail.

The goal is to get the `jenkins.pub` into the jenkins user's authorized_keys file.

To better satisfy this requirement for unknown key file locations,  I can change the `src:` value to a variable and pre-define the variable to what I have now and allow you to provide an extra variable to over-ride the default for local keys.

`ansible-playbook base_ci_update.yaml -e "keys=/home/rocha/.ssh/jenkins.pub"`

Where the keys would be defined be default as:

```
var:
   # By default, the ansible login user passes the jenkins.pub key.
   # Copy this same key into the jenkins user when using Openstack
   keys: "/home/{{ ansible_user }}/.ssh/authorized_keys"
```
I will go ahead and make that change so it is easier to create images locally.